src/lms-ds-loader/edfi_lms_ds_loader/df_to_db.py

```diff
@@ -115,3 +115,21 @@ def upload_assignments(
 
     if not submissions_type_df.empty:
         _upload_assignment_submission_types(db_adapter, submissions_type_df)
+
+
+def upload_section_associations(
+    db_adapter: MssqlLmsOperations, section_associations_df: pd.DataFrame
+) -> None:
+
+    TABLE = Table.SECTION_ASSOCIATION
+    columns = list(section_associations_df.columns)
+
+    _prepare_staging_table(db_adapter, section_associations_df, TABLE)
+    db_adapter.insert_new_records_to_production_for_section_and_user(
+        TABLE,
+        columns
+    )
+    db_adapter.copy_updates_to_production(TABLE, columns)
+    db_adapter.soft_delete_from_production(TABLE, _get_source_system(section_associations_df))
```

Comment:
Is there an opportunity to refactor this and `_upload_assignments` to reduce duplicated code?

Reply:
I've got a nice refactor of this, through it's made testing with mocks a little more interesting...

Reply:
Was able to simplify and share with assignments, along with test simplification.